### PR TITLE
add sitemap.xml and tutorial how to do it

### PR DESCRIPTION
**Pull request by Slashek**
Please note - the metatag are configurable, I've just used `skip_sitemap` as example name, but we might want to use another name

### CODE REVIEW

File: marketplace_builder/views/pages/tutorials/seo/dynamic-sitemap.liquid [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+{% raw %}
+<?xml version="1.0" encoding="UTF-8"?>
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:image="http://www.google.com/schemas/sitemap-image/1.1" xmlns:video="http://www.google.com/schemas/sitemap-video/1.1">
+  {%- query_graph 'sitemap_pages', result_name: g -%}
+  {% for page in g.pages.results %}
+    {% if page.slug == '/' %}{% assign slug = page.slug %}{% else %}{% assign slug = '/' | append: page.slug %}{% endif %}
+    <url>
+      <loc>https://{{ context.location.host }}{{ slug }}</loc>
+      <changefreq>{{ page.metadata.sitemap.changefreq | default: 'monthly' }}</changefreq>
+      <priority>{{ page.metadata.sitemap.changefreq | default: 1 }}</priority>
```
Comment (Slashek) [inline marketplace_builder/views/pages/tutorials/seo/dynamic-sitemap.liquid]:
```suggestion
      <priority>{{ page.metadata.sitemap.priority | default: 1 }}</priority>
```

File: marketplace_builder/views/pages/tutorials/seo/dynamic-sitemap.liquid [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      <loc>https://{{ context.location.host }}{{ slug }}</loc>
+      <changefreq>{{ page.metadata.sitemap.changefreq | default: 'monthly' }}</changefreq>
+      <priority>{{ page.metadata.sitemap.priority | default: 1 }}</priority>
+    </url>
+  {% endfor %}
+</urlset>
+{% endraw %}
+```
+
+Using this template, you are able to explicitly exclude any Page from the sitemap.xml by including `skip_sitemap` key to a `metaga` property of a page. Please note you can also use `metatag` to configure any sitemap properties like changefreq and priority. Please note, the example above already uses static caching for 1 hour - you might want to remove it for development.
```
Comment (Slashek) [inline marketplace_builder/views/pages/tutorials/seo/dynamic-sitemap.liquid]:
```suggestion
Using this template, you are able to explicitly exclude any Page from the sitemap.xml by including `skip_sitemap` key to a `metatag` property of a Page. Please note you can also use `metatag` to configure any sitemap properties like changefreq and priority. Please note, the example above already uses static caching for 1 hour - you might want to remove it for development.
```

**Comment by diana-lakatos** [comment]
Thank you! Reviewed and edited. 
Closes #336

**Comment by Slashek** [comment]
thx :)

File: marketplace_builder/views/pages/sitemap.xml.liquid [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+slug: sitemap
+layout_name: ""
+static_cache:
+  expire: 3600
+metadata:
+  skip_sitemap:
+---
+<?xml version="1.0" encoding="UTF-8"?>
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:image="http://www.google.com/schemas/sitemap-image/1.1" xmlns:video="http://www.google.com/schemas/sitemap-video/1.1">
+  {%- query_graph 'sitemap_pages', result_name: g -%}
```
Comment (pavelloz) [inline marketplace_builder/views/pages/sitemap.xml.liquid]:
```suggestion
  {%- query_graph 'sitemap_pages', result_name: 'g' -%}
```

File: marketplace_builder/views/pages/sitemap.xml.liquid [hunk 4]
```diff
@@ -0,0 +1,20 @@
+---
+slug: sitemap
+layout_name: ""
+static_cache:
+  expire: 3600
+metadata:
+  skip_sitemap:
+---
+<?xml version="1.0" encoding="UTF-8"?>
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:image="http://www.google.com/schemas/sitemap-image/1.1" xmlns:video="http://www.google.com/schemas/sitemap-video/1.1">
```
Comment (pavelloz) [inline marketplace_builder/views/pages/sitemap.xml.liquid]: I dont know what those google thingies are, but even google on their sitemap guide does not mention it.
```suggestion
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
```

Sources:
https://www.sitemaps.org/protocol.html
https://support.google.com/webmasters/answer/183668?hl=en

File: marketplace_builder/views/pages/sitemap.xml.liquid [hunk 5]
```diff
@@ -0,0 +1,20 @@
+---
+slug: sitemap
+layout_name: ""
+static_cache:
+  expire: 3600
+metadata:
+  skip_sitemap:
```
Comment (pavelloz) [inline marketplace_builder/views/pages/sitemap.xml.liquid]:
```suggestion
  skip_sitemap: true
```

File: marketplace_builder/views/pages/tutorials/seo/dynamic-sitemap.liquid [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+static_cache:
+  expire: 3600
+metadata:
+  skip_sitemap:
+---
+```
+```liquid
+{% raw %}
+<?xml version="1.0" encoding="UTF-8"?>
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:image="http://www.google.com/schemas/sitemap-image/1.1" xmlns:video="http://www.google.com/schemas/sitemap-video/1.1">
```
Comment (pavelloz) [inline marketplace_builder/views/pages/tutorials/seo/dynamic-sitemap.liquid]:
```suggestion
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
```

File: marketplace_builder/views/pages/tutorials/seo/dynamic-sitemap.liquid [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+Create `views/pages/sitemap.xml.liquid` page. Use the GraphQL query defined above to iterate dynamically over all pages. 
+
+```yaml
+---
+slug: sitemap
+layout_name: ""
+static_cache:
+  expire: 3600
+metadata:
+  skip_sitemap:
```
Comment (pavelloz) [inline marketplace_builder/views/pages/tutorials/seo/dynamic-sitemap.liquid]:
```suggestion
  skip_sitemap: true
```

File: marketplace_builder/views/pages/tutorials/seo/dynamic-sitemap.liquid [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+---
+```
+```liquid
+{% raw %}
+<?xml version="1.0" encoding="UTF-8"?>
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:image="http://www.google.com/schemas/sitemap-image/1.1" xmlns:video="http://www.google.com/schemas/sitemap-video/1.1">
+  {%- query_graph 'sitemap_pages', result_name: g -%}
```
Comment (pavelloz) [inline marketplace_builder/views/pages/tutorials/seo/dynamic-sitemap.liquid]:
```suggestion
  {%- query_graph 'sitemap_pages', result_name: 'g' -%}
```

File: marketplace_builder/views/pages/tutorials/seo/dynamic-sitemap.liquid [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+## Steps
+
+Creating a dynamic sitemap.xml is a two-step process:
+
+1. Prepare GraphQL query
+2. Setup page
+
+### Step 1: Prepare GraphQL query
+
+Create a GraphQL query which fetches all pages except for the ones that are explicitly excluded via the proper metatag:
```
Comment (pavelloz) [inline marketplace_builder/views/pages/tutorials/seo/dynamic-sitemap.liquid]:
```suggestion
Create a GraphQL query which fetches all pages except for the ones that are explicitly excluded via the proper metadata key:
```

File: marketplace_builder/views/pages/sitemap.xml.liquid [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+---
+<?xml version="1.0" encoding="UTF-8"?>
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:image="http://www.google.com/schemas/sitemap-image/1.1" xmlns:video="http://www.google.com/schemas/sitemap-video/1.1">
+  {%- query_graph 'sitemap_pages', result_name: g -%}
+  {% for page in g.pages.results %}
+    {% if page.slug == '/' %}{% assign slug = page.slug %}{% else %}{% assign slug = '/' | append: page.slug %}{% endif %}
+    <url>
+      <loc>https://{{ context.location.host }}{{ slug }}</loc>
+      <changefreq>{{ page.metadata.sitemap.changefreq | default: 'monthly' }}</changefreq>
+      <priority>{{ page.metadata.sitemap.changefreq | default: 1 }}</priority>
```
Comment (pavelloz) [inline marketplace_builder/views/pages/sitemap.xml.liquid]:
```suggestion
      <priority>{{ page.metadata.sitemap.priority | default: 1 }}</priority>
```

**Comment by pavelloz** [comment]
couple suggestions/typo fixes

**Comment by Slashek** [comment]
thx @pavelloz ❤️

**Pull request merged**